core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -41,8 +45,11 @@ import org.apache.spark.util.{RpcUtils, SerializableBuffer, ThreadUtils, Utils}
  * (spark.deploy.*).
  */
 private[spark]
-class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: RpcEnv)
-  extends ExecutorAllocationClient with SchedulerBackend with Logging
+class CoarseGrainedSchedulerBackend(
+    scheduler: TaskSchedulerImpl,
+    val rpcEnv: RpcEnv,
+    hadoopDelegationTokenManager: Option[HadoopDelegationTokenManager])
```

Comment:
I'm a little torn on having this as a constructor argument. It seems cleaner at first, but it kinda make the constructor of sub-classes (like the mesos one) kinda ugly.

How about having a `protected val hadoopDelegationTokenManager = None` and overriding it where needed? That makes initialization in the sub-class more readable.